test/System/IntegrationTests/DynamoApplicationTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using Dynamo.Applications;
+using NUnit.Framework;
+
+namespace IntegrationTests
+{
+    public class DynamoApplicationTests
+    {
+        [Test]
+        public void MakeModelLoadsASMFromValidPath()
+        {
+            var versions = new List<Version>(){
+                    new Version(224, 4, 0),
+                    new Version(224, 0, 1),
+                    new Version(223, 0, 1),
+                    new Version(225, 0, 0) };
+
+
+            //go get a valid asm path.
+            var locatedPath = string.Empty;
+            var coreDirectory = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+
+            DynamoShapeManager.Utilities.GetInstalledAsmVersion2(versions, ref locatedPath, coreDirectory);
```

Comment:
So for purposes of this test, this could still be doing a recursive lookup of ASM in the product installations.

Reply:
yes, I just wanted the test to pass on any devs machine - BUT a better test would be to then copy the asm dlls somewhere else (like temp directory) and then load them!
what do you think?

Reply:
that could be slow or potentially run into access issues on the test machines though.

Reply:
thinking abit more, I'm inclined to leave this as is - for the purposes of the test like you said, it doesn't really matter that we do the recursive lookup as long as we can assert that the asm path we provided was the asm which was loaded.